File: tests/agent_workloads/adapters/interactive_terminal.py
import os
import subprocess
import sys


if len(sys.argv) != 2:
    raise SystemExit("usage: interactive_terminal INPUT_ROOT")

task = os.environ["JET_CORPUS_TASK"]
script_name = "terminal_closed.sh" if task == "interactive-terminal-closed" else "terminal_session.sh"
answers = "" if task == "interactive-terminal-closed" else "Ada\nblue\n"
completed = subprocess.run(
    ["script", "-qfec", f"sh {script_name}", "/dev/null"],
    cwd=sys.argv[1],
    input=answers,
    text=True,
    capture_output=True,
    timeout=5,
    check=False,
)
if completed.returncode != 0:
    raise SystemExit(completed.stderr or "terminal command failed")
output = completed.stdout
if task == "interactive-terminal-closed":
    if "closed" not in output:
        raise SystemExit("closed terminal did not return")
    print("terminal=pty\nclosed=ok\nexit=0")
else:
    if not all(marker in output for marker in ("Name: ", "Hello Ada", "Choice blue")):
        raise SystemExit("terminal dialogue markers missing")
    print("terminal=pty\nresize=ok\nprompt=ok\nreply=ok")

File: tests/agent_workloads/adapters/interactive_terminal.mjs
import { spawnSync } from "node:child_process";

if (process.argv.length !== 3) {
  throw new Error("usage: interactive_terminal INPUT_ROOT");
}

const task = process.env.JET_CORPUS_TASK;
const scriptName = task === "interactive-terminal-closed" ? "terminal_closed.sh" : "terminal_session.sh";
const input = task === "interactive-terminal-closed" ? "" : "Ada\nblue\n";
const result = spawnSync("script", ["-qfec", `sh ${scriptName}`, "/dev/null"], {
  cwd: process.argv[2],
  input,
  encoding: "utf8",
  timeout: 5000,
});
if (result.error) throw result.error;
if (result.status !== 0) throw new Error(result.stderr || "terminal command failed");
if (task === "interactive-terminal-closed") {
  if (!result.stdout.includes("closed")) throw new Error("closed terminal did not return");
  console.log("terminal=pty\nclosed=ok\nexit=0");
} else {
  for (const marker of ["Name: ", "Hello Ada", "Choice blue"]) {
    if (!result.stdout.includes(marker)) throw new Error("terminal dialogue markers missing");
  }
  console.log("terminal=pty\nresize=ok\nprompt=ok\nreply=ok");
}
